src/Data/Currency.purs

```diff
@@ -38,45 +54,37 @@ parseCentsFromMicroDollars = parseCentsFromNumber <<< flip (/) 10000.0
 centsToMaybeInt :: Cents -> Maybe Int
 centsToMaybeInt = Int.fromString <<< BigInt.toString <<< unwrap
 
--- | Will attempt to parse cents from a string representing a dollar amount. It will
--- strip any trailing cents beyond the hundredths place. WARNING: Do not use this on
--- a string meant to represent cents! It will overstate by 100x.
 parseCentsFromDollarStr :: String -> Maybe Cents
-parseCentsFromDollarStr str = Cents <$> case split (Pattern ".") str of
-  -- There is no decimal; treat as a full dollar string
-  [ dollars ] -> bigIntIs64Bit =<< dollarsPlace dollars
-
-  -- There is one decimal; truncate the cents to 2 places and
-  -- add them to the dollars
-  [ dollars, cents ] ->
-    bigIntIs64Bit
-    =<< (+) <$> dollarsPlace dollars <*> BigInt.fromString (take 2 $ cents <> "0")
+parseCentsFromDollarStr = map Cents <<< parseCurrencyFromDollarStr 100
 
-  -- Unable to parse
-  otherwise -> Nothing
+parseMicroDollarsFromDollarStr :: String -> Maybe MicroDollars
+parseMicroDollarsFromDollarStr = map MicroDollars <<< parseCurrencyFromDollarStr 1000000
 
+-- | Will attempt to parse cents from a string representing a dollar amount. It will
+-- strip any trailing cents beyond the hundredths place. WARNING: Do not use this on
+-- a string meant to represent cents! It assumes non-decimal numbers are whole dollars.
+parseCurrencyFromDollarStr :: Int -> String -> Maybe BigInt
+parseCurrencyFromDollarStr multiplier str =
+  case splitDecimal $ stripCommas str of
+    [ whole ] ->
+      convertCurrency whole
+
+    [ whole, d ] ->
+      let decimal = padRight 2 "0" d
+      in convertCurrency whole `fPlus` (convertCurrency decimal `bigDiv` 100)
+
+    _ ->
+      Nothing
   where
-    -- Expects only the dollars place, no cents. Cents will overstate
-    -- by 100x! Verifies within 64 bit bounds.
-    dollarsPlace :: String -> Maybe BigInt
-    dollarsPlace s
-      | null s = Nothing
-      | otherwise = pure
-          <<< (*) (BigInt.fromInt 100)
-          =<< BigInt.fromString
-          =<< cleanDollars s
-
-    cleanDollars :: String -> Maybe String
```

Comment:
I think the `cleanDollars` value was preventing us from attempting to parse malformed strings like `"1,2,3"`. Do we want to get rid of this behavior? If we **don't** want to get rid of this behavior, I suggest we add some test cases that will catch this in the future. If we do want to get rid of this behavior, ignore this comment.

Since I don't know whether this change is intentional, I defer to you and the others to make the call here.

Reply:
I didn't even think about that. But as far as i've seen, this parsing function seems to assume that the string is in a somewhat acceptable dollar format already. e.g, if you pass in a whole number string like "123", it assumes its 123$, even if the string was not actually dollars. But I guess I can adjust that depending on if we merge this or not